src/func/Microsoft.Health.Fhir.Ingest.Host/Startup.cs

```diff
@@ -0,0 +1,60 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using EnsureThat;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            EnsureArg.IsNotNull(builder, nameof(builder));
+
+            var instrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+
+            if (instrumentationKey != null)
+            {
+                var configDescriptor = builder.Services.SingleOrDefault(tc => tc.ServiceType == typeof(TelemetryConfiguration));
```

Comment:
Would you be able to detail the issue you were hitting with prior suggested line of code?  I am find continuing as is but I think I might want to take a look.  Thanks!

Reply:
I am thinking that the serviceProvider doesn't get built correctly or I just don't understand what it is doing. 

I hit this error: System.InvalidOperationException: 'Unable to resolve service for type 'Microsoft.Azure.WebJobs.Script.IEnvironment' while attempting to activate 'Microsoft.Azure.WebJobs.Script.Configuration.ApplicationInsightsLoggerOptionsSetup'.'

with:

var serviceProvider = builder.Services.BuildServiceProvider();
var tconfig = serviceProvider.GetService(typeof(TelemetryConfiguration));
 
 or

var serviceProvider = builder.Services.BuildServiceProvider();
var tconfig = serviceProvider.GetService<TelemetryConfiguration>();